packages/phpstan-latte-rules/tests/Rules/LatteCompleteCheckRule/LatteCompleteCheckRuleTest.php

```diff
@@ -30,31 +30,39 @@ public function provideData(): Iterator
     {
         // tests @see \PHPStan\Rules\Methods\CallMethodsRule
         $errorMessage = sprintf('Call to an undefined method %s::missingMethod().', SomeTypeWithMethods::class);
-        yield [__DIR__ . '/Fixture/SomeMissingMethodCall.php', [[$errorMessage, 16]]];
+//        yield [__DIR__ . '/Fixture/SomeMissingMethodCall.php', [[$errorMessage, 12]]];
 
         // tests @see \PHPStan\Rules\Methods\CallMethodsRule
         $errorMessage = sprintf(
             'Parameter #1 $name of method %s::render() expects string, int given.',
             InvalidControlRenderArguments::class
         );
-        yield [__DIR__ . '/Fixture/InvalidControlRenderArguments.php', [[$errorMessage, 17]]];
+        yield [__DIR__ . '/Fixture/InvalidControlRenderArguments.php', [[$errorMessage, 15]]];
 
         yield [__DIR__ . '/Fixture/SkipExistingMethodCall.php', []];
         yield [__DIR__ . '/Fixture/SkipVariableInBlockControl.php', []];
 
-        yield [__DIR__ . '/Fixture/GetTemplateAndReplaceExtension.php', $this->createSharedErrorMessages(20)];
-        yield [__DIR__ . '/Fixture/NoAdditionalPropertyRead.php', $this->createSharedErrorMessages(20)];
-        yield [__DIR__ . '/Fixture/PropertyReadTemplate.php', $this->createSharedErrorMessages(24)];
-        yield [__DIR__ . '/Fixture/RenderWithParameters.php', $this->createSharedErrorMessages(17)];
-        yield [
-            __DIR__ . '/Fixture/TemplateAsVariableAndRenderToStringWithParameters.php',
-            $this->createSharedErrorMessages(20),
-        ];
+        yield [__DIR__ . '/Fixture/GetTemplateAndReplaceExtension.php', $this->createSharedErrorMessages(15)];
+        yield [__DIR__ . '/Fixture/NoAdditionalPropertyRead.php', $this->createSharedErrorMessages(15)];
+        yield [__DIR__ . '/Fixture/PropertyReadTemplate.php', $this->createSharedErrorMessages(19)];
+        yield [__DIR__ . '/Fixture/RenderWithParameters.php', $this->createSharedErrorMessages(15)];
+//        yield [
```

Comment:
Also here I probably lost some context and types of variables because of changing MethodCall to ClassMethod. Any idea how to "refresh" it? Ondrej mentioned `PHPStan\Node\InClassNode::class` and `NodeScopeResolver::processNodes` when I discussed with him few months back. But I don't get it O:-)

Reply:
Types are lost only when nodes are changed. We have to handle this Rector, not in PHPStan rules.
It will be probably same problem as ClassMethod node

Reply:
what do you mean by "handle this Rector"